core/src/main/scala/scala/collection/parallel/RemainsIterator.scala

```diff
@@ -465,7 +466,7 @@ self =>
     def split: Seq[IterableSplitter[S]] = self.split.map { _ map f }
   }
 
-  override def map[S](f: T => S) = new Mapped(f)
+  override def map[S](f: T => S): Mapped[S] = new Mapped[S](f)
```

Comment:
I did this here and elsewhere
```
override def map[S](f: T => S): IterableSplitter[S] = new Mapped(f)
```

Reply:
Edit: I didn't sprinkle casts around. ("Cast no casts.")

Reply:
It doesn't matter one way or the other, does it?

Reply:
I suspect it's better to make the change at definition and not cast at use site. But I didn't publish a master's thesis about it.

Reply:
okay, I get it now. fixed.